src/planner/binder/tableref/bind_table_function.cpp

```diff
@@ -236,11 +238,50 @@ unique_ptr<LogicalOperator> Binder::BindTableFunctionInternal(TableFunction &tab
 	get->named_parameters = named_parameters;
 	get->input_table_types = input_table_types;
 	get->input_table_names = input_table_names;
-	if (table_function.in_out_function && !table_function.projection_pushdown) {
+
+	if ((table_function.in_out_function && !table_function.projection_pushdown) || table_function.with_ordinality) {
 		for (idx_t i = 0; i < return_types.size(); i++) {
 			get->AddColumnId(i);
```

Comment:
This prevents projection pushdown from working if `WITH ORDINALITY` is specified - I think that's undesirable since projection pushdown has a large positive performance effect. Is this necessary here?